src/main/java/org/opensearch/neuralsearch/processor/TextEmbeddingProcessor.java

```diff
@@ -0,0 +1,269 @@
+            treeRes.put(key, sourceAndMetadataMap.get(parentKey));
+        }
+    }
+
+    @SuppressWarnings({ "unchecked" })
+    @VisibleForTesting
+    Map<String, Object> buildVectorOutput(
+        Map<String, Object> knnMap,
+        List<List<Float>> modelTensorList,
+        Map<String, Object> sourceAndMetadataMap
+    ) {
+        IndexWrapper indexWrapper = new IndexWrapper(0);
+        Map<String, Object> result = new LinkedHashMap<>();
+        for (Map.Entry<String, Object> knnMapEntry : knnMap.entrySet()) {
+            String knnKey = knnMapEntry.getKey();
+            Object sourceValue = knnMapEntry.getValue();
+            if (sourceValue instanceof String && StringUtils.isNotBlank(sourceValue.toString())) {
+                List<Float> modelTensor = modelTensorList.get(indexWrapper.index++);
+                result.put(knnKey, modelTensor);
+            } else if (sourceValue instanceof List) {
+                result.put(knnKey, processListOut((List<String>) sourceValue, modelTensorList, indexWrapper));
+            } else if (sourceValue instanceof Map) {
+                processMapTypeVectorOutput(knnKey, sourceValue, modelTensorList, indexWrapper, sourceAndMetadataMap);
+            }
+        }
+        return result;
+    }
+
+    @SuppressWarnings({ "unchecked" })
+    private void processMapTypeVectorOutput(
+        String knnKey,
+        Object sourceValue,
+        List<List<Float>> modelTensorList,
+        IndexWrapper indexWrapper,
+        Map<String, Object> sourceAndMetadataMap
+    ) {
+        if (knnKey == null || sourceAndMetadataMap == null || sourceValue == null) return;
+        if (sourceValue instanceof Map) {
+            for (Map.Entry<String, Object> inputNestedMapEntry : ((Map<String, Object>) sourceValue).entrySet()) {
+                processMapTypeVectorOutput(
+                    inputNestedMapEntry.getKey(),
+                    inputNestedMapEntry.getValue(),
+                    modelTensorList,
+                    indexWrapper,
+                    (Map<String, Object>) sourceAndMetadataMap.get(knnKey)
+                );
+            }
+        } else if (sourceValue instanceof String) {
+            sourceAndMetadataMap.put(knnKey, modelTensorList.get(indexWrapper.index++));
+        } else if (sourceValue instanceof List) {
+            sourceAndMetadataMap.put(knnKey, processListOut((List<String>) sourceValue, modelTensorList, indexWrapper));
+        }
+    }
+
+    private List<Map<String, List<Float>>> processListOut(
+        List<String> sourceValue,
+        List<List<Float>> modelTensorList,
+        IndexWrapper indexWrapper
+    ) {
+        List<Map<String, List<Float>>> numbers = new ArrayList<>();
+        for (String strSourceValue : sourceValue) {
+            if (StringUtils.isNotBlank(strSourceValue)) {
+                numbers.add(ImmutableMap.of(LIST_TYPE_NESTED_MAP_KEY, modelTensorList.get(indexWrapper.index++)));
+            }
```

Comment:
if we remove blank strings here, then the order vector lists will be wrong. Example:
```
Input
["This is"," ","Hello world"]

Output vectors in document:
[{"vector":[1.0,2.0]},{"vector":[4.0,3.0]}]
```
This seems to be wrong. I also think we add an empty vector list here that will cause the exception. Can we go with vector list of 0.0 values ?

@jmazanec15 do we see any issue with 0.0 value? Will is cause any impact on the K-NN? 

@ylwu-amzn if we pass a blank string or a string with spaces to ML Algorithms what will happen? will it fail? I think ans is no. want to confirm.